include/radio_types.h

```diff
@@ -2124,9 +2138,43 @@ typedef struct radio_cell_info_1_5 {
         RadioCellInfoNr_1_5 nr RADIO_ALIGNED(8);
         RadioCellInfoCdma_1_2 cdma RADIO_ALIGNED(8);
     } info RADIO_ALIGNED(8);
-} RADIO_ALIGNED(8) RadioCellInfo_1_5; /* Since 1.5.0 */
+} RADIO_ALIGNED(8) RadioCellInfoRatSpecificInfo_1_5; /* Since 1.5.4 */
+G_STATIC_ASSERT(sizeof(RadioCellInfoRatSpecificInfo_1_5) == 192);
+
+typedef struct radio_cell_info_1_5 {
+    guint8 registered RADIO_ALIGNED(1);
+    gint32 timeStampType RADIO_ALIGNED(4);
+    guint64 timeStamp RADIO_ALIGNED(8);
+    RADIO_CELL_CONNECTION_STATUS connectionStatus RADIO_ALIGNED(4);
+    RadioCellInfoRatSpecificInfo_1_5 ratSpecificInfo RADIO_ALIGNED(8);
+} RADIO_ALIGNED(8) RadioCellInfo_1_5; /* Since 1.5.4 */
 G_STATIC_ASSERT(sizeof(RadioCellInfo_1_5) == 216);
 
+typedef struct radio_physical_channel_config {
+    RADIO_CELL_CONNECTION_STATUS connectionStatus RADIO_ALIGNED(4);
+    gint32 cellBandwidthDownlink RADIO_ALIGNED(4);
+} RADIO_ALIGNED(4) RadioPhysicalChannelConfig; /* Since 1.5.4 */
+G_STATIC_ASSERT(sizeof(RadioPhysicalChannelConfig) == 8);
+
+typedef struct radio_frequency_info {
+    guint8 frequencyInfoType RADIO_ALIGNED(4); /* RADIO_FREQUENCY_INFO_TYPE */
+    union {
+        RADIO_FREQUENCY_RANGE range RADIO_ALIGNED(4);
+        gint32 channelNumber RADIO_ALIGNED(4);
+    } info RADIO_ALIGNED(4);
+} RADIO_ALIGNED(4) RadioFrequencyInfo; /* Since 1.5.4 */
+G_STATIC_ASSERT(sizeof(RadioFrequencyInfo) == 8);
+
+typedef struct radio_physical_channel_config_1_4 {
+    RADIO_CELL_CONNECTION_STATUS connectionStatus RADIO_ALIGNED(4);
+    gint32 cellBandwidthDownlink RADIO_ALIGNED(4);
```

Comment:
I suggest
 ```c
RadioPhysicalChannelConfig base;
```
instead of these two fields, to mirror the definitions in `types.hal`/`types.h`. That sort of practice helps code reuse, e.g. allows to move the code handling the base structure into a separate function and call it from more than one place.

Reply:
Fixed